search/search_quality/samples_generation_tool/samples_generation_tool.cpp

```diff
@@ -0,0 +1,404 @@
+#include "search/categories_cache.hpp"
+#include "search/mwm_context.hpp"
+#include "search/reverse_geocoder.hpp"
+#include "search/search_quality/helpers.hpp"
+#include "search/search_quality/sample.hpp"
+
+#include "indexer/categories_holder.hpp"
+#include "indexer/classificator_loader.hpp"
+#include "indexer/data_source.hpp"
+#include "indexer/feature_algo.hpp"
+#include "indexer/ftypes_matcher.hpp"
+#include "indexer/search_string_utils.hpp"
+
+#include "coding/file_name_utils.hpp"
+
+#include "platform/local_country_file.hpp"
+#include "platform/local_country_file_utils.hpp"
+#include "platform/platform.hpp"
+
+#include "geometry/mercator.hpp"
+
+#include "base/macros.hpp"
+#include "base/string_utils.hpp"
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <cstdlib>
+#include <ctime>
+#include <fstream>
+#include <limits>
+#include <map>
+#include <random>
+#include <set>
+#include <string>
+#include <vector>
+
+#include "3party/gflags/src/gflags/gflags.h"
+
+using namespace search;
+using namespace std;
+
+DEFINE_string(data_path, "", "Path to data directory (resources dir)");
+DEFINE_string(mwm_path, "", "Path to mwm files (writable dir)");
+DEFINE_string(out_buildings_path, "buildings.json", "Path to output samples file");
+DEFINE_string(out_cafes_path, "cafes.json", "Path to output samples file");
```

Comment:
Fix the docs for buildings and cafes, please.

Reply:
done